kotishell/src/tokenize/tokenize_a.c

```diff
@@ -0,0 +1,90 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        ::::::::            */
+/*   tokenize_a.c                                       :+:    :+:            */
+/*                                                     +:+                    */
+/*   By: elenavoronin <elnvoronin@gmail.com>          +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2023/06/29 16:20:23 by elenavoroni   #+#    #+#                 */
+/*   Updated: 2023/07/13 18:54:20 by dkolodze      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "tokenize.h"
+
+static void	s_tk_whitespace(t_tk_so_far *so_far, char *s, t_tk_result *result)
+{
+	if (so_far->token.type != TK_WHITESPACE && so_far->token.length == 0)
```

Comment:
There can be 2 of same token in a row for pipes in our shell; maybe replace `so_far->token.length == 0` with `so_far->token.type == TK_EMPTY`

Reply:
agreed!